src/lib/eslint/BaseEslintEngine.ts

```diff
@@ -206,7 +207,7 @@ export abstract class BaseEslintEngine implements RuleEngine {
 				}
 
 				// get run-config for the engine and add to config
-				Object.assign(config, await this.strategy.getRunConfig(engineOptions));
+				Object.assign(config, deepCopy(await this.strategy.getRunConfig(engineOptions)));
```

Comment:
Changed all instances of Object.assign to use deepCopy. Only the `BaseEslintEngine` instance demonstrated the bug during unit tests, but any usage could suffer from the same issue.